gcovr/writer/cobertura.py

```diff
@@ -74,14 +74,52 @@ def print_cobertura_report(covdata: CovData, output_file, options):
             ),
         )
         c = etree.Element("class")
-        # The Cobertura DTD requires a methods section, which isn't
-        # trivial to get from gcov (so we will leave it blank)
-        etree.SubElement(c, "methods")
-        lines = etree.SubElement(c, "lines")
 
+        sorted_lines = sorted(data.lines)
+        iter_reversed_sorted_lines = iter(reversed(sorted_lines))
+        methods = []
+        for function in sorted(data.functions, key=lambda f: data.functions[f].lineno, reverse=True):
+            function_cov = data.functions[function]
+            elem = _method_element(function_cov)
+            methods.append(elem)
+
+            lines = etree.Element("lines")
+            elem.append(lines)
+            function_branch = CoverageStat(0, 0)
+            function_linenos = []
+            for lineno in iter_reversed_sorted_lines:
+                if lineno >= function_cov.lineno:
+                    function_linenos.insert(0, lineno)
+                else:
+                    break
```

Comment:
This seems to assume that every line belongs to a function, and that there's no code between functions. Perhaps it would be better to avoid making such guesses, and just generate an empty `<lines/>` element in each method section.

Reply:
But then the method element has no information about coverage.